lib/s3.py

```diff
@@ -71,15 +80,28 @@ class BotoS3Client(RigorS3Client):
 
 	def __init__(self, config, bucket, credentials=None):
 		super(BotoS3Client, self).__init__(config, bucket, credentials)
-		connection_args = list()
-		if credentials:
-			connection_args.append(config.get(credentials, 'aws_access_key_id'))
-			connection_args.append(config.get(credentials, 'aws_secret_access_key'))
-		self._conn = S3Connection(*connection_args)
-		self.bucket = self._conn.get_bucket(bucket)
+		if 'boto3' in sys.modules:
```

Comment:
I feel like having this check only in init and using that to store the result is less likely to lead to potential inconsistencies (i.e. boto is used at first, then some other library pulls in boto3 and all of a sudden the same rigors3 client is doing boto3 things—a strange runtime effect).

Arguably it might even be cleaner to use the init to monkeypatch the shadow methods over the original ones, removing the conditionals from the original methods.